04/modules/WeatherGUI.js

```diff
@@ -0,0 +1,167 @@
+            this.loadApiData();
+            this.clearErrorsList();
+        }
+
+        this.renderErrors(errors);
+    }
+
+    validateInputsData(errors) {
+        const fields = [
+            { name: 'lat', label: 'latitude', required: true, pattern: '^-?([0-9]{1,2}|1[0-7][0-9]|180)(\.[0-9]{1,10})$', },
+            { name: 'lng', label: 'longitude', required: true, pattern: '^-?([0-9]{1,2}|1[0-7][0-9]|180)(\.[0-9]{1,10})$', },
+        ];
+
+        fields.forEach(field => {
+            const { name, label, required, pattern } = field;
+
+            const inputValue = this.formEl.elements[name].value;
+
+            if (required) {
+                if (this.isElementEmpty(inputValue)) {
+                    errors.push(`Pole ${label.toUpperCase()} jest wymagane.`);
+                }
+            }
+            if (inputValue.length > 0 && pattern) {
+                if (this.isNotValidPattern(pattern, inputValue)) {
+                    errors.push(`Wprowadź poprawną wartość w polu ${label} (np. "52.232222").`);
+                }
+            }
+        })
+    }
+
+    isNotValidPattern(pattern, inputValue) {
+        const reg = new RegExp(pattern);
+        if (!reg.test(inputValue)) { return true }
+    }
+
+    sendToApi(data) {
+        this.api.getGeoCoords(data);
+    }
+
+    loadApiData() {
+        this.api.loadData()
+            .then(data => this.insertData(data))
+            .catch(err => console.error(err));
+    }
+
+    insertData(weatherData) {
+        const { lon, lat, temp, city_name, weather: { description } } = weatherData;
+        const [parentEl] = this.weatherInfoElement.children;
+        const [latEl, lngEl, cityEl, descriptionEl, temperatureEl] = parentEl.children;
+
+        const infoElementsArr = [
+            { element: latEl, text: lat },
+            { element: lngEl, text: lon },
+            { element: cityEl, text: `${city_name}` },
+            { element: descriptionEl, text: description },
+            { element: temperatureEl, text: temp },
+        ];
+
+        this.renderWeatherData(infoElementsArr);
+    }
+
+    renderWeatherData(infoElementsArr) {
+        infoElementsArr.forEach(infoElement => {
+            const { element, text } = infoElement;
+
+            element.innerText = text;
+        });
+    }
+
+    renderErrors(errors) {
+        if (!this.isElementEmpty(errors)) {
+            this.renderFormErrors(errors);
+            this.renderSectionError();
+        }
+    }
+
+    renderFormErrors(errors) {
+        const errorsList = this.createList(errors, 'form__errors-list', 'form__errors-list-item');
+        errorsList.style.color = 'red';
+        errorsList.style.listStyle = 'none';
+
+        this.clearErrorsList();
+
+        this.formEl.insertBefore(errorsList, this.formEl.lastElementChild);
+    }
+
+    renderSectionError() {
+        const errorEl = this.createElement('p', 'weather__error');
+
+        this.weatherInfoElement.children[0].style.display = 'none';
```

Comment:
Zawsze jak gdzieś masz [0] należy się stanowić nie wykorzystać destrukturyzacji i tutaj warto - bo określimy co to za element. Będzie czytelniej.